docs/operator-manual/applicationset/Generators-Matrix.md

```diff
@@ -104,18 +104,80 @@ Finally, the Matrix generator will combine both sets of outputs, and produce:
 ```
 (*The full example can be found [here](https://github.com/argoproj/argo-cd/tree/master/applicationset/examples/matrix).*)
 
+## Using Parameters from one child generator in another child generator
+
+The Matrix generator allows using the parameters generated by one child generator inside another child generator. 
+Below is an example that uses a git-files generator in conjunction with a cluster generator.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-git
+spec:
+  generators:
+    # matrix 'parent' generator
+    - matrix:
+        generators:
+          # git generator, 'child' #1
+          - git:
+              repoURL: https://github.com/argoproj/applicationset.git
+              revision: HEAD
+              files:
+                - path: "examples/git-generator-files-discovery/cluster-config/**/config.json"
+          # cluster generator, 'child' #2
+          - clusters:
+              selector:
+                matchLabels:
+                  argocd.argoproj.io/secret-type: cluster
+                  kubernetes.io/environment: '{{path.basename}}'
+  template:
+    metadata:
+      name: '{{name}}-guestbook'
+    spec:
+      project: default
+      source:
+        repoURL: https://github.com/argoproj/applicationset.git
+        targetRevision: HEAD
+        path: "examples/git-generator-files-discovery/apps/guestbook"
+      destination:
+        server: '{{server}}'
+        namespace: guestbook
+```
+Here is the corresponding folder structure for the git repository used by the git-files generator:
+
+```
+├── apps
+│   └── guestbook
+│       ├── guestbook-ui-deployment.yaml
+│       ├── guestbook-ui-svc.yaml
+│       └── kustomization.yaml
+├── cluster-config
+│   └── engineering
+│       ├── dev
+│       │   └── config.json
+│       └── prod
+│           └── config.json
+└── git-generator-files.yaml
+```
+In the above example, the `{{path.basename}}` parameters produced by the git-files generator will resolve to `dev` and `prod`.
+In the 2nd child generator, the label selector with label `kubernetes.io/environment: {{path.basename}}` will resolve with the values produced by the first child generator's parameters (`kubernetes.io/environment: prod` and `kubernetes.io/environment: dev`). 
+
+So in the above example, clusters with the label `kubernetes.io/environment: prod` will have only prod-specific configuration (ie. `prod/config.json`) applied to it, wheres clusters
+with the label `kubernetes.io/environment: dev` will have only dev-specific configuration (ie. `dev/config.json`)
+
 ## Restrictions
 
 1. The Matrix generator currently only supports combining the outputs of only two child generators (eg does not support generating combinations for 3 or more).
-1. You should specify only a single generator per array entry, eg this is not valid:
+2. You should specify only a single generator per array entry, eg this is not valid:
```

Comment:
please don't update the numbers

Reply:
Should I change them all back to "1." ? I thought that was an error?

Reply:
Yeah, it's like that, check main Readme.md it will get rendered in ordered way

Reply:
Sounds good, will change it

Reply:
Ok, should be done